### PR TITLE
🐛 Source FB Marketing: deprecate `INSIGHTS_DAYS_PER_JOB` from connector's specification.

### LINKED ISSUE #7848: Clarify the setup fields for Source Facebook Marketing
## Tell us about the problem you're trying to solve
The fields in the Facebook setup are not organized in a logic order. End Date is optional, but listed first. Users don't understand which fields required or optional. 

## Describe the solution you’d like
1. Start Date should be listed first
2. End Date should be listed after Start Date, and the label changed to End Date (Optional)
3. Custom Insights can be relabelled as “Custom Insights (Optional)” with help text as to what it is. A user sees this configuration and assumed they need to fill something in. Users are confused on whether they should fill something in. They should understand that the connection will work even if they do not fill in anything here. 
4. Account ID should show an example value (eg. 111111111111) in the help text. One user thought it had to start with “cus_” or “act_”  because other tools accept it this way or it's displayed in the Facebook UI this way.

### LINKED ISSUE #8027: Source FB Marketing: Deprecate the `Insights Days per job` parameter
## Tell us about the problem you're trying to solve
A question from a user: 

> For the Facebook Marketing API, how do I know if I have a lot of data? Put another way, how could I spot if I have too much data for a given time period and thus calibrate my time period to suit the size of my data?

image

IMO exposing this option was a tactical error on our part. Like described in our new [UX handbook](https://docs.airbyte.io/connector-development/ux-handbook#input-configuration), input configuration like this is an implementation detail and should not be offloaded to the user. We should just remove it and use the default value of 1. 

## Acceptance Criteria
1. option is removed from spec.json (in a backwards compatible way)
2. a sensible value is chosen by default

### PR DESCRIPTION
**Pull request by bazarnov**
## What
#8027

## How
- edited `source.py`
- corrected `Readme.md`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/8234)

### CODE REVIEW

File: airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/source.py [hunk 1]
```diff
@@ -342,12 +352,13 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
...
         """
-        authenticator = PayPalOauth2Authenticator(config)
+        #authenticator = PayPalOauth2Authenticator(config)
+        authenticator = PayPalAuthenticator(config).get_auth()
 
         # Try to get API TOKEN
-        token = authenticator.get_access_token()
-        if not token:
-            return False, "Unable to fetch Paypal API token due to incorrect client_id or secret"
+        #token = authenticator.get_access_token()
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/source.py:359]: this seems unrelated to this change?

Comment (bazarnov) [reply airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/source.py:359]: 100% correct, it's a mistake.

Comment (bazarnov) [reply airbyte-integrations/connectors/source-paypal-transaction/source_paypal_transaction/source.py:359]: reverted this up to master state.

**Review by sherifnada — APPROVED** [review]
LGTM - @keu as fb connector author, any feedback on this?

**Comment by keu** [comment]
@sherifnada @bazarnov forcing it to be 1 day can make the sync process 7x,.. 10x times slower, this was the reason to introduce it in the first place. The user was using 7days interval, when we were using 1 day.

File: airbyte-integrations/connectors/source-facebook-marketing/acceptance-test-config.yml [hunk 2]
```diff
@@ -14,14 +14,15 @@ tests:
   basic_read:
     - config_path: "secrets/config.json"
       configured_catalog_path: "integration_tests/configured_catalog.json"
-      timeout_seconds: 600
+      timeout_seconds: 3600
```
Comment (keu) [inline airbyte-integrations/connectors/source-facebook-marketing/acceptance-test-config.yml:17]: @sherifnada you can see it here for example

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py [hunk 3]
```diff
@@ -56,43 +60,49 @@ class ConnectorConfig(BaseModel):
...
+    include_deleted: bool = Field(
+        title="Include Deleted", default=False, description="Include data from deleted campaigns, ads, and adsets.", order=4
+    )
 
     insights_lookback_window: int = Field(
-        default=28,
-        description="The attribution window for the actions",
-        minimum=0,
-        maximum=28,
+        title="Insights Lookback Window", default=28, description="The attribution window for the actions", minimum=0, maximum=28, order=5
```
Comment (keu) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:99]: this formatting looks awful, where is this coming from?

**Comment by sherifnada** [comment]
@keu fair enough. What's the path forward for deprecating this option, then?

**Comment by bazarnov** [comment]
> @keu fair enough. What's the path forward for deprecating this option, then?

@keu we can set the default value to 7 instead of 1.

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py (hunk 3 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py]: Corrected.

File: airbyte-integrations/connectors/source-facebook-marketing/acceptance-test-config.yml (hunk 2 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-facebook-marketing/acceptance-test-config.yml]: Set this to 1200, and INSIGHTS_DAYS_PER_JOB to 7, instead of 1.

**Review by annalvova05 — APPROVED** [review]

**Review by midavadim — APPROVED** [review]

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py (hunk 3 above)
Comment (keu) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py]: @bazarnov thanks, 
I mean no offense, just was curious how is it possible that the build passed successfully and the code doesn't look formatted with black (usually black produces code with good readability). Is this some IDE tool?

**Comment by keu** [comment]
> @keu fair enough. What's the path forward for deprecating this option, then?

@sherifnada @bazarnov I can't give a perfect solution right now, if I had one, I wouldn't add such a configuration option. The problem with any default value is that the user or us can't tune the value. 

If we set this value too big we might have too many fails on accounts with a large amount of data and vise versa.
Some connector (don't remember the name) that Dmytro implemented has dynamic slices that the adopt amount of data it fetches to the percentage of fails. Maybe we can decrease the size of the interval each time AsyncJob fails. The challenge here is new jobs for missing subslices.

let say we have jobs with interval = 5
slices/jobs:
ddddd ddddd ddddd

the second slice failed, we retry with interval 2, but 3rd slice is still running, and we need to add missing jobs for the rest of 2nd slice

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py [hunk 4]
```diff
@@ -40,59 +40,88 @@
 
 logger = logging.getLogger("airbyte")
 
+# deprecated from ConnectorSpec, using default value of 1 = 1 day.
+# Related Issue here: https://github.com/airbytehq/airbyte/issues/8027
+INSIGHTS_DAYS_PER_JOB: int = 7
```
Comment (avida) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:45]: why it says 1 in comment and has value of 7?

Comment (bazarnov) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:45]: Here is why: https://github.com/airbytehq/airbyte/pull/8234#issuecomment-979488000

Comment (bazarnov) [reply airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:45]: Oh, my bad. Corrected.

> 

@keu What are the root causes of the possible job failure with date_slices? I've tested using the values of 300 and 1000 for the `insights days per job` - all are successfully processed on our test account. 
Therefore we can try to stay on value of 3 or 5 as a default value. WDYT?

**Comment by bazarnov** [comment]
> > @keu fair enough. What's the path forward for deprecating this option, then?
> 
> @sherifnada @bazarnov I can't give a perfect solution right now, if I had one, I wouldn't add such a configuration option. The problem with any hard-coded default value is that the user or us can't tune the value.
> 
> If we set this value too big we might have too many fails on accounts with a large amount of data and vise versa. Some connector (don't remember the name) that Dmytro implemented has dynamic slices that adopt the amount of data it fetches to the percentage of fails. Maybe we can decrease the size of the interval each time AsyncJob fails. The challenge here is new jobs for missing subslices.
> 
> let say we have jobs with interval = 5 slices/jobs: ddddd ddddd ddddd
> 
> the second slice failed, we retry with interval 2, but 3rd slice is still running, and we need to add missing jobs for the rest of 2nd slice

@keu What are the root causes of the possible job failure with date_slices? I've tested using the values of 300 and 1000 for the `insights days per job` - all are successfully processed on our test account. 
Therefore we can try to stay on value of 3 or 5 as a default value. WDYT?

**Comment by keu** [comment]
> > > @keu fair enough. What's the path forward for deprecating this option, then?
> > 
> > 
> > @sherifnada @bazarnov I can't give a perfect solution right now, if I had one, I wouldn't add such a configuration option. The problem with any hard-coded default value is that the user or us can't tune the value.
> > If we set this value too big we might have too many fails on accounts with a large amount of data and vise versa. Some connector (don't remember the name) that Dmytro implemented has dynamic slices that adopt the amount of data it fetches to the percentage of fails. Maybe we can decrease the size of the interval each time AsyncJob fails. The challenge here is new jobs for missing subslices.
> > let say we have jobs with interval = 5 slices/jobs: ddddd ddddd ddddd
> > the second slice failed, we retry with interval 2, but 3rd slice is still running, and we need to add missing jobs for the rest of 2nd slice
> 
> @keu What are the root causes of the possible job failure with date_slices? I've tested using the values of 300 and 1000 for the `insights days per job` - all are successfully processed on our test account. Therefore we can try to stay on value of 3 or 5 as a default value. WDYT?

Since we have a very little amount of data testing on our test account is not even close to the production accounts of users.
I don't know an exact root cause, my guess from my experience and singer implementation the more data we need to fetch the more chance that job will fail (probably memory limitation or something) so decreasing `insights days per job` to 1 day theoretically should help (and helps), but not always (my guess it can be solved by querying fewer aggregation fields).

@avida is currently investigating this issue with the account provided by our customer (Cart).

**Comment by sherifnada** [comment]
@bazarnov let's postpone deprecating the inisght days per job until https://github.com/airbytehq/airbyte/pull/8385 is merged

**Comment by keu** [comment]
The deprecation was done in the scope of #8385 because it also touches the logic.
The only missing part is the re-ordering of columns. I think it makes sense to move it there as well and close this PR.

**Comment by bazarnov** [comment]
Closed with reference to this: https://github.com/airbytehq/airbyte/pull/8234#issuecomment-1016248376
